compiler/circle-mpqsolver/src/bisection/DepthParameterizer.cpp

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) 2022 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "DepthParameterizer.h"
+
+namespace mpqsolver
+{
+namespace bisection
+{
+
+/**
+ * @brief compute maximal distance from graph inputs to graph nodes along with min/max values of
+ * distance and return status of computation (Assumes graph has no cycles)
+ */
+ParameterizerResult compute_depth(const luci::Module *module, NodeDepthType &nodes_depth,
+                                  float &min_depth, float &max_depth)
+{
+  if (module == nullptr)
+    return ParameterizerResult::FAILURE;
+
+  if (module->size() != 1)
+    return ParameterizerResult::FAILURE;
+
+  auto graph = module->graph(0);
+  if (!graph)
+    return ParameterizerResult::FAILURE;
+
+  auto nodes = graph->nodes();
+  uint32_t nodes_size = nodes->size();
+  std::set<std::string> input_names;
```

Comment:
I think `input_names` is not used. Maybe inserted for debugging purpose?

Reply:
Yes. It was done for debugging.I'll remove them. Thank you.

Reply:
Fixed